lua/neo-tree/sources/common/hijack_cursor.lua

```diff
@@ -0,0 +1,38 @@
+local events = require("neo-tree.events")
+local log = require("neo-tree.log")
+
+local M = {}
+
+local hijack_cursor_handler = function()
+    if vim.o.filetype ~= "neo-tree" then
+        return
+    end
+    local source = vim.api.nvim_buf_get_var(0, "neo_tree_source")
+    local state = require("neo-tree.sources.manager").get_state(source)
+    if state == nil then
+      return
+    end
+    local winid = state.winid
+    if vim.api.nvim_get_current_win() == winid then
+      local node = state.tree:get_node()
+      log.debug("Cursor moved in tree window, hijacking cursor position")
+      local cursor = vim.api.nvim_win_get_cursor(0)
+      local row = cursor[1]
+      local current_line = vim.api.nvim_get_current_line()
```

Comment:
I think this could be:
```suggestion
      local current_line = cursor[0]
```

Reply:
I think that you meant `cursor[2]` but that returns only the column position. Here I need to fetch the text for the current line.

Reply:
You're right, I was very confused. I was thinking that you were looking for the current line number for some reason, and also that tables were zero indexed.